cli/cmd/generate_terraform_test.go

```diff
@@ -0,0 +1,215 @@
+	"github.com/hashicorp/hc-install/releases"
+	"github.com/stretchr/testify/assert"
+)
+
+// Test various scenarios for detection and/or installation of Terraform binary
+func TestGenerationTfInstallation(t *testing.T) {
+	if os.Getenv("CI") == "true" {
+		t.Run("forced installation should installed expected version", func(t *testing.T) {
+			toggleNonInteractive()
+			defer toggleNonInteractive()
+			dir, err := ioutil.TempDir("", "tf-install-test")
+			if err != nil {
+				panic(err)
+			}
+			defer os.RemoveAll(dir)
+
+			// Force install
+			tf, err := LocateOrInstallTerraform(true, dir)
+			installedTempPath := filepath.Dir(tf.ExecPath())
+			defer os.RemoveAll(installedTempPath)
+
+			// Validate install works via version command and that installed version is correct
+			version, _, _ := tf.Version(context.Background(), true)
+			assert.Equal(t, installTerraformVersion, version.String())
+			assert.NoError(t, err)
+		})
+		t.Run("existing terraform version should be located and used, if version is new enough", func(t *testing.T) {
+			toggleNonInteractive()
+			defer toggleNonInteractive()
+			dir, err := ioutil.TempDir("", "tf-install-test")
+			if err != nil {
+				panic(err)
+			}
+			defer os.RemoveAll(dir)
+
+			// Install some version that should be detected
+			installer := &releases.ExactVersion{
+				Product: product.Terraform,
+				Version: version.Must(version.NewVersion("1.0.10")),
+			}
+
+			installPath, err := installer.Install(context.Background())
+			if err != nil {
+				panic(err)
+			}
+			installedTempPath := filepath.Dir(installPath)
+			defer os.RemoveAll(installedTempPath)
+
+			// Update path to find the new version
+			oldPath := os.Getenv("PATH")
+			os.Setenv("PATH", fmt.Sprintf("%s:%s", installedTempPath, oldPath))
+			defer os.Setenv("PATH", oldPath)
+
+			// Locate or install should find version 1.0.10
+			tf, err := LocateOrInstallTerraform(false, dir)
+			version, _, _ := tf.Version(context.Background(), true)
+			assert.Equal(t, "1.0.10", version.String())
+			assert.Equal(t, installedTempPath, filepath.Dir(tf.ExecPath()))
+			assert.NoError(t, err)
+		})
+		t.Run("installed version of terraform that is too old gets ephemeral installation", func(t *testing.T) {
+			toggleNonInteractive()
+			defer toggleNonInteractive()
+			dir, err := ioutil.TempDir("", "tf-install-test")
+			if err != nil {
+				panic(err)
+			}
+			defer os.RemoveAll(dir)
+
+			// Install some version that should be detected
+			installer := &releases.ExactVersion{
+				Product: product.Terraform,
+				Version: version.Must(version.NewVersion("0.13.7")),
+			}
+
+			installPath, err := installer.Install(context.Background())
+			if err != nil {
+				panic(err)
+			}
+			installedTempPath := filepath.Dir(installPath)
+			defer os.RemoveAll(installedTempPath)
+
+			// Update path to find the new version
+			oldPath := os.Getenv("PATH")
+			os.Setenv("PATH", fmt.Sprintf("%s:%s", installedTempPath, oldPath))
```

Comment:
Does this work on Windows? 🤔

Reply:
No, it did not; so I updated here: https://github.com/lacework/go-sdk/pull/642/commits/4026aaaebf8b28b285140d176ab41303adb17215#diff-6cc7b544129624856ba0189d1ac4c8bb7995e5706282949c4bcc44d8b82c06a8R106

There's a little logic at the top of that test that figures out the appropriate env separator depending on platform. That allows this set of tests to pass in windows.